modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java

```diff
@@ -202,16 +202,26 @@ private RatedRequest mutateTestItem(RatedRequest original) {
         List<String> summaryFields = original.getSummaryFields();
         String templateId = original.getTemplateId();
 
-        int mutate = randomIntBetween(0, 7);
+        int mutate = randomIntBetween(0, 5);
         switch (mutate) {
             case 0:
                 id = randomValueOtherThan(id, () -> randomAsciiOfLength(10));
                 break;
             case 1:
-                int size = randomValueOtherThan(testRequest.size(), () -> randomInt());
-                testRequest = new SearchSourceBuilder();
-                testRequest.size(size);
-                testRequest.query(new MatchAllQueryBuilder());
+                if (testRequest != null) {
+                    int size = randomValueOtherThan(testRequest.size(), () -> randomInt());
+                    testRequest = new SearchSourceBuilder();
+                    testRequest.size(size);
+                    testRequest.query(new MatchAllQueryBuilder());
+                } else {
+                    if (randomBoolean()) {
+                        params = new HashMap<>();
+                        params.putAll(params);
```

Comment:
This looks odd, can you recheck this is what you intended? Local params vs. the ones defined earlier?